pkg/apis/hive/v1/syncset_types.go

```diff
@@ -259,6 +245,20 @@ type SyncSetSpec struct {
 	ClusterDeploymentRefs []corev1.LocalObjectReference `json:"clusterDeploymentRefs"`
 }
 
+// SyncSetCommonStatus is common status for SyncSets and SelectorSyncSets
+type SyncSetCommonStatus struct {
+	// ObservedGeneration is the generation of the syncset that has been observed by the syncset controller.
+	ObservedGeneration int64 `json:"observedGeneration"`
+
+	// Resources is the list of resources effected by the syncset.
+	// +optional
+	Resources []SyncResourceReference `json:"resources,omitempty"`
```

Comment:
This is here for two reasons.
1. I had the idea that it would be beneficial to put conditions on the syncsets themselves if they are bad rather than trying to apply the syncset to each cluster. This makes it easier to alert on a bad syncset, rather than having to deduce that the syncset is bad based on failure to apply to every cluster.
2. In the syncsetinstance controller, there was a discovery call made to the target cluster as part of decoding the resource. I initially moved the decoding to the syncset controller with the thought that I could amortize the cost of decoding. This did not bear out, as the discovery was just removed from the decoding instead.

In summary, there is not any real benefit from storing the resource identifications (or the observed generation) in the syncset status.

Reply:
Thanks for the explanation. If we can remove it now before it gets baked in, so much the better.

Reply:
To me, this also helps to identify syncsets with malformed resources before attempting to sync them. We're taking blobs and extracting basic information about them (type, name, namespace). I'm in favor of leaving it in.

Reply:
It is a nicety, but I do not think that it warrants the added complexity. It requires a separate controller and makes the clustersync controller dependent on that controller working. I am in favor of removing it.

Reply:
As a user, I can get the same information with the following.
```
oc get syncset my-syncset -ojson | jq '.spec.resources[]? | [.apiVersion, .kind, .metadata.namespace, .metadata.name]'
```